### ISSUE TITLE
[pickers] Open picker when clicking `Space` on field

### ISSUE THREAD

**Issue by LukasTy** [mui/mui-x#15299]
Created from: https://github.com/mui/mui-x/issues/14587#issuecomment-2459608283.

> We are closely following the guidelines of https://www.w3.org/WAI/ARIA/apg/patterns/dialog-modal/examples/datepicker-dialog/ but [native date input](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/date) element does indeed open on `Space` click.

**Comment by LukasTy** [comment]
- Should be worked after https://github.com/mui/mui-x/issues/14787 is done.
- Look into applying it on the v7 (input) behavior.
- Test with `playwright` and `@testing-library/user-event` to see if entering "2022 04 17" causes the spaces to open the picker
  - if that's a problem, consider relying on the search query (if it not empty - do not open on <kbd>Space</kbd> click)
